commit 109c32ffd89d64dd99a775f2f50443bee38b63e9
Author: Matt Redfearn <matt.redfearn@mips.com>
Date:   Thu Nov 24 17:32:45 2016 +0000

    MIPS: Add support for ARCH_MMAP_RND_{COMPAT_}BITS
    
    arch_mmap_rnd() uses hard-coded limits of 16MB for the randomisation
    of mmap within 32bit processes and 256MB in 64bit processes. Since v4.4
    other arches support tuning this value in /proc/sys/vm/mmap_rnd_bits.
    Add support for this to MIPS.
    
    Set the minimum(default) number of bits randomisation for 32bit to 8 -
    which with 4k pagesize is unchanged from the current 16MB total
    randomness. The minimum(default) for 64bit is 12bits, again with 4k
    pagesize this is the same as the current 256MB.
    
    This patch is necessary for MIPS32 to pass the Android CTS tests, with
    the number of random bits set to 15.
    
    Signed-off-by: Matt Redfearn <matt.redfearn@imgtec.com>
    Reviewed-by: Kees Cook <keescook@chromium.org>
    Cc: Paul Gortmaker <paul.gortmaker@windriver.com>
    Cc: Daniel Cashman <dcashman@android.com>
    Cc: Andrew Morton <akpm@linux-foundation.org>
    Cc: linux-mips@linux-mips.org
    Cc: kernel-hardening@lists.openwall.com
    Cc: linux-kernel@vger.kernel.org
    Patchwork: https://patchwork.linux-mips.org/patch/14617/
    Signed-off-by: Ralf Baechle <ralf@linux-mips.org>

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 9a7730219c93..6ed1a0af33e9 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -14,6 +14,8 @@ config MIPS
 	select HAVE_PERF_EVENTS
 	select PERF_USE_VMALLOC
 	select HAVE_ARCH_KGDB
+	select HAVE_ARCH_MMAP_RND_BITS if MMU
+	select HAVE_ARCH_MMAP_RND_COMPAT_BITS if MMU && COMPAT
 	select HAVE_ARCH_SECCOMP_FILTER
 	select HAVE_ARCH_TRACEHOOK
 	select HAVE_CBPF_JIT if !CPU_MICROMIPS
@@ -3073,6 +3075,20 @@ config MMU
 	bool
 	default y
 
+config ARCH_MMAP_RND_BITS_MIN
+	default 12 if 64BIT
+	default 8
+
+config ARCH_MMAP_RND_BITS_MAX
+	default 18 if 64BIT
+	default 15
+
+config ARCH_MMAP_RND_COMPAT_BITS_MIN
+       default 8
+
+config ARCH_MMAP_RND_COMPAT_BITS_MAX
+       default 15
+
 config I8253
 	bool
 	select CLKSRC_I8253